src/manifest/dbtProject.ts

```diff
@@ -1005,18 +1006,20 @@ select * from renamed
     }
     const columns: Record<string, ColumnMetaData> = {};
     Object.entries(node.columns).forEach(([k, v]) => {
-      columns[k.toLowerCase()] = v;
+      columns[getColumnNameByCase(k)] = v;
     });
 
     for (const c of columnsFromDB) {
-      const existing_column = columns[c.column.toLowerCase()];
+      const existing_column = columns[getColumnNameByCase(c.column)];
       if (existing_column) {
-        existing_column.data_type = existing_column.data_type || c.dtype;
+        existing_column.data_type = (
+          existing_column.data_type || c.dtype
+        )?.toLowerCase();
```

Comment:
Should the lower case config be used here as well?

Reply:
for data_type it will always be in lowercase from now